modules/social_features/social_user/src/Plugin/AdvancedQueue/JobType/UserMailQueueJob.php

```diff
@@ -156,42 +190,65 @@ public function process(Job $job) {
    *   The email content from the storage entity.
-   * @param string $display_name
-   *   In case of anonymous users a display name will be given.
+   * @param string|null $display_name
+   *   In the case of anonymous users, a display name will be given.
+   * @param bool $include_mail_footer
+   *   TRUE if need to include the email footer.
    */
-  protected function sendMail(string $user_mail, string $langcode, QueueStorageEntity $mail_params, $display_name = NULL) {
-    $subject = $mail_params->get('field_subject')->getValue();
-    $message = $mail_params->get('field_message')->getValue();
-    $reply_to = $mail_params->get('field_reply_to')->getValue();
-    $reply = NULL;
-
-    if (!empty($reply_to)) {
-      $reply = $reply_to[0]['value'];
-    }
+  protected function sendMail(string $user_mail, string $langcode, QueueStorageEntity $mail_params, string $display_name = NULL, bool $include_mail_footer = FALSE): void {
+    $subject = $mail_params->get('field_subject')->value;
+    $body = $mail_params->get('field_message')->value;
+    $reply_to = $mail_params->get('field_reply_to')->getString();
+
+    if ($include_mail_footer && $this->moduleHandler->moduleExists('social_email_broadcast')) {
+      $params = ['subject' => $subject, 'body' => $body];
+
+      $settings_link = Link::fromTextAndUrl($this->t('email notification settings', [], ['langcode' => $langcode]),
+        Url::fromRoute('social_email_broadcast.user_edit_page')->setAbsolute())->toString();
+
+      // Construct the render array for email.
+      $notification = [
+        '#theme' => 'directmail',
+        '#notification' => $params['body'],
+        '#notification_settings' => $this->t('You receive community updates and announcements emails according to your @settings', [
+          '@settings' => $settings_link,
+        ], ['langcode' => $langcode]),
+      ];
 
-    $context = [
-      'subject' => $subject[0]['value'],
-      'message' => $message[0]['value'],
-    ];
+      $params['body'] = $this->renderer->renderPlain($notification);
+      $params['reply-to'] = $reply_to;
 
-    if ($display_name) {
-      $context['display_name'] = $display_name;
+      // Attempt sending mail.
+      $this->mailManager->mail(
+        'social_email_broadcast',
+        'social_email_broadcast',
+        $user_mail,
+        $langcode,
+        $params
+      );
     }
+    else {
+      $context = ['subject' => $subject, 'message' => $body];
 
-    // Attempt sending mail.
-    $this->mailManager->mail(
-      'system',
-      'action_send_email',
-      $user_mail,
-      $langcode,
-      ['context' => $context],
-      $reply,
-    );
+      if ($display_name) {
+        $context['display_name'] = $display_name;
+      }
+
+      // Attempt sending mail.
+      $this->mailManager->mail(
+        'system',
+        'action_send_email',
+        $user_mail,
+        $langcode,
+        ['context' => $context],
+        $reply_to,
+      );
```

Comment:
Given the difference in logic, I'm wondering if this queue shouldn't be split up as well? It seems like we're still intertwining the notification system and the organiser broadcast system here.

Reply:
I don't think we have intertwining the notification system and email broadcast system there (at least, there are no mentions of a notification system)

Reply:
Ah sorry, I thought `action_send_email` was `activity_send_email`. 👍 

The split between using different mail types depending on whether a module is enabled feels difficult, but I can see how there's not really a way around that at the moment.